src/tests/fs_tests.rs

```diff
@@ -342,6 +342,314 @@ pub mod fs_tests {
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        //Creating a directory.
+        assert_eq!(cage.mkdir_syscall("/testdir", S_IRWXA), 0);
+        let fd = cage.open_syscall("/testdir", O_RDWR, S_IRWXA);
+
+        //Checking if passing the created directory to
+        //`mmap_syscall()` correctly results in `The `fildes`
+        //argument refers to a file whose type is not
+        //supported by mmap` error.
+        assert_eq!(
+            cage.mmap_syscall(0 as *mut u8, 5, PROT_READ | PROT_WRITE, MAP_PRIVATE, fd, 0),
+            -(Errno::EACCES as i32)
+        );
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
+    #[test]
+    pub fn ut_lind_fs_mmap_invalid_fildes() {
+        //acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        //Creating a regular file with `O_RDWR` flag
+        //making it valid for any mapping and then
+        //closing it, thereby making the obtained
+        //filede scriptor invalid because no other
+        //file is opened after it.
+        let flags: i32 = O_TRUNC | O_CREAT | O_RDWR;
+        let filepath = "/mmapTestFile1";
+        let fd = cage.open_syscall(filepath, flags, S_IRWXA);
+        assert_eq!(cage.close_syscall(fd), 0);
+
+        //Checking if passing the invalid file descriptor
+        //correctly results in `Invalid file descriptor` error.
+        assert_eq!(
+            cage.mmap_syscall(0 as *mut u8, 5, PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0),
+            -(Errno::EBADF as i32)
+        );
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
+    #[test]
+    pub fn ut_lind_fs_munmap_zerolen() {
+        //acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        //Creating a regular file with `O_RDWR` flag
+        //making it valid for any mapping.
+        let flags: i32 = O_TRUNC | O_CREAT | O_RDWR;
+        let filepath = "/mmapTestFile1";
+        let fd = cage.open_syscall(filepath, flags, S_IRWXA);
+        //Writing into that file's first 9 bytes.
+        assert_eq!(cage.write_syscall(fd, str2cbuf("Test text"), 9), 9);
+
+        //Checking if passing 0 as `len` to `munmap_syscall()`
+        //correctly results in 'The value of len is 0` error.
+        assert_eq!(
+            cage.munmap_syscall(0 as *mut u8, 0),
+            -(Errno::EINVAL as i32)
+        );
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
```

Comment:
Can you check with the tests for other cases of `munmap`, where we unmap a valid memory address space for a process?